### DISCUSSION TITLE
Allow inclusion of images from the clipboard

### DISCUSSION THREAD

**Discussion by henningsway** [quarto-dev/quarto-cli#4385] [Q&A] [answered]
Make it easy to copy + paste images into a quick report. (This is obviously not super reproducible and maybe this should show up in the report by default (the fact that it wasn't created by code))

I often take screenshots of pictures and it would be nice to be able to just "drag" them into the report and have the extension handle the file conversion and inclusion of the image.

Sth along the lines of https://github.com/Toniiiio/imageclipr, ideally also for vscode.

**Comment by mcanouil** [comment, marked as answer]
I do not think this is a Quarto CLI feature, but eventually an IDE feature.

Paste from clipboard requires additional work that in my opinion should be done by the other, _i.e._, save the image as an image.

Note that for plain markdown document, it's possible to use drag&drop, see <https://code.visualstudio.com/Docs/languages/markdown#_drag-and-drop-to-insert-links-and-images>.

It's not by default enabled for Quarto document since they are not by default recognise as markdown documents.

It's definitely something that could be enabled in Quarto VSCode extension (<https://github.com/quarto-dev/quarto/issues>).

EDIT: "markdown paste" extension, see <https://marketplace.visualstudio.com/items?itemName=telesoho.vscode-markdown-paste-image>

**Comment by henningsway** [comment]
I didn't know about native vscode drag&drop functionality for images, that's quite nice already, thank you.

I feel that inserting say a screenshot of a website into a presentation is currently a little too much work:

1. take screenshot
2. save to folder next to report
3. include link

if this could be handled automatically with some reasonable defaults it would make the process easier. :)

**Reply by mcanouil** [reply]
If you are using VSCode you could use the "Markdown Paste" extension.

**Reply by henningsway** [reply]
Super helpful, thx!

**Reply by mistune** [reply]
@mcanouil This one really works well. Thank you!

**Comment by mcanouil** [comment]
I am going to close this as resolved.
